sklearn/feature_selection/from_model.py

```diff
@@ -201,7 +222,15 @@ def _get_support_mask(self):
         scores = _get_feature_importances(estimator)
         self.threshold_ = _calculate_threshold(estimator, scores,
                                                self.threshold)
-        return scores >= self.threshold_
+        mask = np.zeros_like(scores, dtype=bool)
+        if self.max_features == 'all':
+            self.max_features = scores.size
+        candidate_indices = np.argsort(-scores,
+                                       kind='mergesort')[:self.max_features]
+        mask[candidate_indices] = True
+        if self.threshold is not None:
```

Comment:
only if you changed the behavior, which you shouldn't have.. otherwise it will use the default threshold.